.semgrep/adapter/parse-bid-type-check.yml

```diff
@@ -0,0 +1,8 @@
+rules:
+  - id: parse-bid-type-check
+    message: The current implementation follows an anti-pattern, assumes that if there is a multi-format request, the media type defaults to $ORTBTYPE. Prebid server expects the media type to be explicitly set in the adapter response. Therefore, we strongly recommend implementing a pattern where the adapter server sets the [MType](https://github.com/prebid/openrtb/blob/main/openrtb2/bid.go#L334) field in the response to accurately determine the media type for the impression.
```

Comment:
What is the issue with this check? If the adapter's bid response returns the bid type as a string, this function is good for verifying it is a valid bid type and setting it to the correct type. If there is a reason to discourage using this function, we should also add a comment to the function to explain when it should not be used.

Reply:
Good point. Question related to bid type check: what if bidder doesn't return `mtype`? Currently only 4 adapters handle `mtype`.

Reply:
We are only adding this as an warning. We are recommending adapters to use MType instead of using request impressions or sending bid type in bid ext. Currently, adapters are using `ParseBidType` or `imp.<Banner|Video|Native|Audio>` check to determine the bidType as we do using switch case. The way this function is written, it cannot be used with bid.MType directly as MType makes use of `MarkupType`. So this check is adding a suggestion to use bid.MType instead of setting it in bid.ext in response or using request impression to identify the bid type.

Reply:
Right, right, I understand what you are trying to do here. I'm just not sure it can help you with reviews a lot.

Reply:
Heads Up: I'm working through an insanely large PR which will remove the BidType in favor of MarkupType for every adapter. Not sure if that will invalidate this rule's logic.

Reply:
Ooof

Reply:
Let's keep this rule for now. We can update/delete it once adapters start using MType